lightning/src/ln/outbound_payment.rs

```diff
@@ -82,15 +82,37 @@ impl PendingOutboundPayment {
 			attempts.count += 1;
 		}
 	}
-	fn is_retryable_now(&self) -> bool {
-		if let PendingOutboundPayment::Retryable { retry_strategy, attempts, .. } = self {
-			return retry_strategy.is_retryable_now(&attempts)
+	fn auto_retries_remaining_now(&self) -> usize {
+		match self {
+			PendingOutboundPayment::Retryable { retry_strategy: Some(strategy), attempts, .. } => {
+				strategy.retries_remaining_now(&attempts)
+			},
+			_ => 0,
+		}
+	}
+	fn is_retryable_now(&self, pending_paths: usize) -> bool {
+		match self {
+			PendingOutboundPayment::Retryable { retry_strategy: None, .. } => {
+				// We're handling retries manually, we can always retry.
```

Comment:
I thought we were planning to drop support for manual retries. Is there a use case for them? It's a bit confusing that `fail_htlc` checks `payment.get().auto_retries_remaining_now() > 0` instead of `is_retryable_now()`

Reply:
Discussed offline, it seems like we should remove support for manual retries pre-release. Since IIUC there's no reason to keep it in this PR, could we remove it, since it adds a bit of review burden?

Reply:
Oh, I thought we could remove the bits from this PR without touching `InvoicePayer`, let's leave it then

Reply:
I thought we didn't care about that in between releases since we're removing `InvoicePayer` anyway, but maybe that'd be a bridge too far. In any case, seems like too small of a complexity diff to be worth too much debate

Reply:
Maybe it'd break a bunch of tests and other unpleasantness, though.